lab-will/app/entry.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+require('./scss/reset.scss');
+require('./scss/main.scss');
+
+const angular = require('angular');
+const cowsay = require('cowsay-browser');
+
+// Create the entire app
+const cowsayApp = angular.module('cowsayApp', []);
+
+// Create and name the CowsayController
+cowsayApp.controller('CowsayController', ['$log', CowsayController]);
+
+// Define the CowsayController constructor
+function CowsayController($log) {
+  // Debug
+  $log.debug('CowsayController');
+
+  // Properties
+  this.title = 'Welcome to Cowville!';
+  this.history = [];
+
+  // History Stack
+  cowsay.list( (err, cowfiles) => {
+    this.cowfiles = cowfiles;
+    this.current = cowfiles[0];
+  });
```

Comment:
This is not the history stack.  It is a list of all of the available cow forms that is returned by a built in function on cowsay called 'list'.